Src/Particle/AMReX_ArrayOfStructs.H

```diff
@@ -57,36 +57,36 @@ public:
     int numTotalParticles () const { return m_data.size(); }
 
     void setNumNeighbors (int num_neighbors)
-    { 
+    {
         auto nrp = numRealParticles();
         m_num_neighbor_particles = num_neighbors;
         resize(nrp + num_neighbors);
     }
 
     int getNumNeighbors () { return m_num_neighbor_particles; }
-    
+
     bool empty () const { return m_data.empty(); }
-    
-    const RealType* data () const { return &(m_data[0].pos(0)); }
+
+    const RealType* data () const { return &(m_data[0].m_rdata.pos[0]); }
     RealType* data () { return &(m_data[0].pos(0)); }
```

Comment:
might need an adjustment for the non-const case as well?

Reply:
The non-const actually works as is, the issue is that the `pos` method of particle returns a reference in the non-const case, but a value in the const case. So when you try to take the address of a value, you get the `rvalue` compile errors. 

Perhaps that version should return `const RealType&`, but we might need to be careful about possible performance implications.

Reply:
ok, thanks for clarifying -- that helps me better understand the const changes as well.